diesel/src/connection/mod.rs

```diff
@@ -379,10 +388,16 @@ where
 ///
 /// This is a separate trait to allow connection implementations to specify
 /// different loading modes via the generic paramater.
-pub trait LoadConnection<B = DefaultLoadingMode>: Connection
-where
-    for<'conn, 'query> Self: ConnectionGatWorkaround<'conn, 'query, Self::Backend, B>,
-{
+pub trait LoadConnection<B = DefaultLoadingMode>: Connection {
+    /// The cursor type returned by [`LoadConnection::load`]
+    ///
+    /// Users should handle this as opaque type that implements [`Iterator`]
+    type Cursor<'conn, 'query>: Iterator<Item = QueryResult<Self::Row<'conn, 'query>>>;
```

Comment:
The `LoadConnection::load` method should be changed so that it returns `QueryResult<Self::Cursor<'conn, 'query>` now. I think that's much easier to understand for potential users + it's the same type as before.